studio/src/mas-copy-dialog.js

```diff
@@ -0,0 +1,425 @@
+                    ...folder,
+                    displayName: folder.name.replace(/-/g, ' ').replace(/\b\w/g, (l) => l.toUpperCase()),
+                    fullPath: folder.path,
+                }));
+
+            this.loading = false;
+        } catch (err) {
+            this.error = 'Failed to load folders';
+            this.loading = false;
+            console.error('Error loading folders:', err);
+        }
+    }
+
+    selectFolder(folder) {
+        this.selectedFolder = folder;
+        this.error = null;
+    }
+
+    validateInputs() {
+        if (!this.selectedFolder) {
+            this.error = 'Please select a destination folder';
+            return false;
+        }
+
+        if (!this.fragmentName?.trim()) {
+            this.error = 'Please enter a name for the fragment';
+            return false;
+        }
+
+        if (!this.aem) {
+            this.error = 'AEM instance not available';
+            return false;
+        }
+
+        if (!this.fragment?.path) {
+            this.error = 'Fragment is missing path property';
+            return false;
+        }
+
+        return true;
+    }
+
+    showSuccessMessage(copiedFragment) {
+        const folderName = this.selectedFolder.displayName;
+
+        if (copiedFragment.renamedTo && copiedFragment.renamedTo !== this.fragment.name) {
+            showToast(
+                `Fragment copied to ${folderName} and renamed to '${copiedFragment.renamedTo}' to avoid conflicts`,
+                'positive',
+            );
+        } else if (copiedFragment.name && copiedFragment.name !== this.fragment.name) {
+            showToast(
+                `Fragment copied to ${folderName} and renamed to '${copiedFragment.name}' to avoid conflicts`,
+                'positive',
+            );
+        } else {
+            showToast(`Fragment copied to ${folderName}`, 'positive');
+        }
+    }
+
+    handleCopyError(error) {
+        this.error = error.message || 'Failed to copy fragment';
+        this.loading = false;
+
+        const errorMessage = error.message || '';
+
+        for (const [, config] of Object.entries(ERROR_TYPES)) {
+            if (config.patterns.some((pattern) => errorMessage.includes(pattern))) {
+                showToast(config.message, 'negative');
+                return;
+            }
+        }
+
+        showToast(`Failed to copy fragment: ${error.message}`, 'negative');
+    }
+
+    async performCopy() {
+        const customName = this.fragmentName.trim();
+        const copiedFragment = await this.aem.sites.cf.fragments.copyToFolder(
+            this.fragment,
+            this.selectedFolder.fullPath,
+            customName !== this.fragment.name ? customName : null,
+            this.selectedLocale,
+        );
+
+        if (!copiedFragment) {
+            throw new Error('Copy operation completed but could not retrieve copied fragment');
+        }
+
+        return copiedFragment;
+    }
+
+    async handleSubmit() {
```

Comment:
during the copy, can you unset originalId, it might create invalid cross surface references which can be hard to troubleshoot.